tests/test_invenio_records_rest.py

```diff
@@ -81,6 +84,18 @@ def create_record(data):
     return pid, record
 
 
+def delete_record(pid, record):
+    with db.session.begin_nested():
+        record.delete()
```

Comment:
@jirikuncar If it is only a concern on code duplication I will call the delete from the REST API instead. It will be cleaner that way.